### ISSUE TITLE
Package `wasm-tools` as WASM Components

### ISSUE THREAD

**Issue by yoshuawuyts** [bytecodealliance/wasm-tools#1240]
`jco` currently maintains [custom wrappers](https://github.com/bytecodealliance/jco/tree/main/crates/wasm-tools-component) around the various `wasm-tools` tools. This is used to expose the functionality provided by `wasm-tools` from JavaScript.

It would be nice if those bindings could be provided by `wasm-tools` directly, rather than being wrapped by `jco`. Presumably other language-specific toolchains may want to also package `wasm-tools`, and being able to share the same package definitions would be great.

I'm intentionally not requesting this be made a priority. But I did want to flag it, since it seems like something this would be good to do _eventually_. Ideally around the time using [warg](https://warg.io) as a distribution mechanism for Components becomes viable [^viability].

Thank you!

[^viability]: I haven't kept up with the state of `warg`, so I don't know how far along it is. My understanding was that it's still pretty early, and it'll be a little while longer before we can use it to manage and distribute Components?

**Comment by calvinrp** [comment]
Yes, it makes sense to have wrappers live in `wasm-tools` and publish the `wasm-tools` components from that repo.

This is on our radar and we want this as well. We are aiming to start distributing components on `warg` registries toward the end of this month (November 2023). I'll report back with updates regarding registry availability.
